gtsfm/frontend/correspondence_augmenter.py

```diff
@@ -0,0 +1,179 @@
+            putative_corr_idxs_to_augment,
+        ) = self._correspondence_generator.generate_correspondences(
+            client=client, images=images, image_pairs=list(two_view_outputs.keys())
+        )
+
+        def verify_putative_correspondences(
+            keypoints_i1: Keypoints,
+            Keypoints_i2: Keypoints,
+            putative_corr_idxs: np.ndarray,
+            camera_intrinsics_i1: gtsfm_types.CALIBRATION_TYPE,
+            camera_intrinsics_i2: gtsfm_types.CALIBRATION_TYPE,
+            i2Ri1: Rot3,
+            i2Ui1: Unit3,
+            triangulation_options: TriangulationOptions,
+            ba_optimizer: BundleAdjustmentOptimizer,
+        ) -> np.ndarray:
+            if i2Ri1 is None or i2Ui1 is None:
+                return np.array([], dtype=np.uint64)
+
+            i2Ti1_prior = PosePrior(
+                value=Pose3(i2Ri1, i2Ui1.point3()),
+                covariance=RELATIVE_POSE_PRIOR_SIGMA,
+                type=PosePriorType.SOFT_CONSTRAINT,
+            )
+
+            _, _, verified_corr_idxs = two_view_ba_utils.bundle_adjust(
+                keypoints_i1=keypoints_i1,
+                keypoints_i2=Keypoints_i2,
+                verified_corr_idxs=putative_corr_idxs,
+                camera_intrinsics_i1=camera_intrinsics_i1,
+                camera_intrinsics_i2=camera_intrinsics_i2,
+                i2Ri1_initial=i2Ri1,
+                i2Ui1_initial=i2Ui1,
+                i2Ti1_prior=i2Ti1_prior,
+                triangulation_options=triangulation_options,
+                ba_optimizer=ba_optimizer,
+            )
+
+            return verified_corr_idxs
+
+        verified_corr_idxs_futures = {
+            (i1, i2): client.submit(
+                verify_putative_correspondences,
+                keypoints_to_augment_list[i1],
+                keypoints_to_augment_list[i2],
+                putative_corr_idxs_to_augment[(i1, i2)],
+                camera_intrinsics[i1],
+                camera_intrinsics[i2],
+                two_view_output[0],
+                two_view_output[1],
+                self._triangulation_options,
+                self._ba_optimizer,
+            )
+            for (i1, i2), two_view_output in two_view_outputs.items()
+        }
+        verified_corr_idxs_dict = client.gather(verified_corr_idxs_futures)
+
+        augmented_keypoints_list: List[Keypoints] = []
+        keypoint_idx_offsets: List[int] = []
+        for keypoints_original, keypoints_new in zip(keypoints_list, keypoints_to_augment_list):
+            keypoints_augmented, offset = self._augment_keypoints(
+                keypoints_original=keypoints_original, keypoints_new=keypoints_new
+            )
+            augmented_keypoints_list.append(keypoints_augmented)
+            keypoint_idx_offsets.append(offset)
+
+        # Merge the corr_idxs
+        results: Dict[Tuple[int, int], TWO_VIEW_OUTPUT] = {}
+        for (i1, i2), two_view_output in two_view_outputs.items():
```

Comment:
Could we share the code block here with the code here? https://github.com/borglab/gtsfm/blob/master/gtsfm/frontend/correspondence_generator/keypoint_aggregator/keypoint_aggregator_unique.py#L43-L56 Is it identical?